packages/http/src/validator/validators/body.ts

```diff
@@ -29,6 +29,30 @@ export function deserializeFormBody(
     return decodedUriParams;
   }
 
+  // if the request body array contains JSON objects with multiple fields (i.e. '{"foo":"value"}, {"foo":"dd","xx":"xx"}'),
+  // depending on the deserialization indicated in the encoding, it's possible we end up with an array of broken JSON
+  // objects that were split on the ',' character, such as [ '{"foo":"value"}', '{"foo":"dd"', '"xx":"xx"}' ]. This function
+  // processes such cases so that complete JSON objects, i.e. [ '{"foo":"value"}', '{"foo":"dd", "xx":"xx"}' ], are handled
+  function parseBrokenJSONArray(inputArray: string[]): string[] {
+    let parsedJSONObjects: any[] = [];
+    let currentJSONObject: string = "";
+  
+    for (let item of inputArray) {
+      // handle the scenario where a JSON object in the encoded array is preceded by a "+", which can occur when 
+      // the user puts a space between JSON array entries, such as '{"foo": "a"}, {"foo":"b"}'
+      item = item.startsWith("+") ? item.substring(1) : item;
```

Comment:
Those cases you mention actually would pass, but you're right if we had a case like {",+foo+bar": ...} this would not pass (with , and + next to each other). This is because the JSON data is getting split on the comma "," , and so this is the case where we'd have items starting with ",". I did some research and I discovered that the inappropriate presence of commas in the input string is due to a bug in decodeURIComponent specifically for application/x-www-form-urlencoded data. In application/x-www-form-urlencoded format, the standard encoding of spaces is the plus sign "+", and plus signs in the input string are encoded as "%2B". The encoding of spaces as plus signs is unique to   application/x-www-form-urlencoded. I discovered that decodeURIComponent incorrectly handles decoding spaces. encodeURIComponent correctly encodes spaces as + (plus signs), and actual plus signs as %2B, but decodeURIComponent only decodes %2B and leaves the +'s that represent spaces encoded as +. This means the result has + signs that are indistinguishable as originally spaces or originally plus signs. Therefore, in line 140 of packages/http/src/validator/validators/body.ts, I replace all + in the encoded string (which must all represent spaces by the standard), with %20, the non-application/x-www-form-urlencoded  encoding of spaces, so that decodeURIComponent processes them correctly and we don't have this issue to begin with.